src/pages/PositionDetails/Contents.tsx

```diff
@@ -70,34 +78,45 @@ export const Contents = ({ position }: Props) => {
       .map((value, index) => (value === '1' ? index + 1 : 0))
       .filter((n) => !!n)
   })
-  const dropdownItems = [
-    {
-      onClick: () => {
-        logger.log('Redeem')
+  const dropdownItems = useMemo(() => {
+    return [
+      {
+        onClick: () => {
+          setValue(id)
+          history.push(`/redeem`)
+        },
+        text: 'Redeem',
       },
-      text: 'Redeem',
-    },
-    {
-      onClick: () => {
-        logger.log('Split')
+      {
+        onClick: () => {
+          setValue(id)
+          history.push(`/split`)
+        },
+        text: 'Split',
       },
-      text: 'Split',
-    },
-  ]
+    ]
+  }, [id, history, setValue])
 
   // TODO: refactoring this to make work wrap and unwrap
   const ERC20Amount = 100
   const ERC1155Amount = 0
 
   React.useEffect(() => {
     try {
-      const tokenSymbol = networkConfig.getTokenFromAddress(collateralToken.id).symbol
-      setCollateralSymbol(tokenSymbol)
+      const token = networkConfig.getTokenFromAddress(collateralToken.id)
+      setCollateralSymbol(token.symbol)
+      setCollateral(token)
```

Comment:
If I change this to only set collateral I have to add some non-null check, maybe a better option for this pattern is to have a dummy token with an empty symbol and address.